docker-hatch

```diff
@@ -34,6 +34,14 @@ while :; do
         -f|--force-rebuild-docker-image)
             CACHE_FLAG='--no-cache'
             ;;
+        # These are flags for hatch commands: https://hatch.pypa.io/latest/cli/reference/#hatch
```

Comment:
I don't think this is required, the way `docker-hatch` work is that it looks at its own flag until we reach the hatch command. It then break from the loop and pass the rest as-is.

If you run:

```
./docker-hatch run -h
```

Then, under the hood it will run: `hatch run -h`.

The only case you can't use hatch flag is if you try to run a hatch without a command.

Say you wanted to run:

```
hatch --help
```

If you run:

```
./docker-hatch --help
```

Then, the `--help` flag will be read by the script. I would simply remove line 21-24 in this script.

Reply:
Maybe I can give this another look, but I couldn't find a way to run with the `-v` flag to get verbose logging while creating environments which was crucial for finding where the pip installs were hanging. The -v was getting swallowed up by our `-v` to use python version. Maybe you're right and I just needed to rename ours to `-P` and the rest would work. I can try again to see if that alone does the job and then possibly remove this.

Reply:
I don't think you need to rename our `-v` flag to `-P`. As long as you pass the hatch `-v` flag after whatever command you wanted to run, it should work.

```
./docker-hatch -v 3.9 COMMAND -v
```

The command above would run `hatch command -v` in a Python 3.9 docker image.

Reply:
Ah, right, I remember now. The problem is our case statement below that flags unknown options and prints the `usage` and exits. So either we put the known hatch flags in here or we remove that block. Maybe the latter is better.

Reply:
as long as you put the hatch flags after the hatch COMMAND, it should just work.

Can you share the command you are trying to run?

Reply:
`./docker-hatch -v env create hf-datasets`

If you put the `-v` after `env`, you get `Error: No such option` because `-v` isn't a flag for the `env` command`, it's a flag for `hatch`.

Reply:
Ah. I see, now I get it. The way `blaze` handle this is with the `--`:

```
blaze run cloud/kaggle/kernels/tools:vmtd -- --binary $ABUSE_BINARY_MD5HASH
```

We could do:

```
./docker-hatch -- -v env create hf-datasets.
```

And in our logic reading flag in a loop, break if we get `--`.